packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/OpenStackVirtualMachinesRow.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { TableCell } from 'src/modules/Providers/utils';
+
+import { ResourceFieldFactory, RowProps } from '@kubev2v/common';
+import { VSphereVM } from '@kubev2v/types';
+import { Tr } from '@patternfly/react-table';
+
+import { VMCellProps } from './components';
+import { vmsFieldsMetadataFactory } from './ProviderVirtualMachines';
+import { defaultCellRenderers, renderTd, VmData } from './ProviderVirtualMachinesRow';
+
+export const openStackVmFieldsMetadataFactory: ResourceFieldFactory = (t) => [
+  ...vmsFieldsMetadataFactory(t),
```

Comment:
a. please don't refactor at this stage, we will refactor after the lists are working
b. please move the list page, in our past lists, we put the metadata factory in the file we define the list

```suggestion
  {
    resourceFieldId: 'name',
    jsonPath: '$.name',
    label: t('Name'),
    isVisible: true,
    isIdentity: true, // Name is sufficient ID when Namespace is pre-selected
    filter: {
      type: 'freetext',
      placeholderLabel: t('Filter by name'),
    },
    sortable: true,
  },
  {
    resourceFieldId: 'concerns',
    jsonPath: '$.concerns',
    label: t('Concerns'),
    isVisible: true,
    sortable: true,
    filter: {
      type: 'enum',
      primary: true,
      placeholderLabel: t('Concerns'),
      values: EnumToTuple({ Critical: 'Critical', Warning: 'Warning', Information: 'Information' }),
    },
  },```